integration/helpers.go

```diff
@@ -1863,3 +1863,36 @@ func getLocalIP() (string, error) {
 	}
 	return "", trace.NotFound("No non-loopback local IP address found")
 }
+
+// eventually checks a given condition at regular intervals and returns
+// true if the condition is met within the specified duration.
+func eventually(cond func() bool, interval time.Duration, duration time.Duration) bool {
+	return check(cond, true, interval, duration)
+}
+
+// never checks a given condition at regular intervals and returns
+// false if the condition is met within the specified duration.
+func never(cond func() bool, interval time.Duration, duration time.Duration) bool {
+	return check(cond, false, interval, duration)
+}
+
+// check checks a given condition at regular intervals and returns
+// the value of shouldPassCheck if the condition is met within the specified
+// duration.
+func check(cond func() bool, shouldPassCheck bool, interval time.Duration, duration time.Duration) bool {
+	ctx, cancel := context.WithTimeout(context.Background(), duration)
+	defer cancel()
+	for {
+		select {
+		case <-ctx.Done():
+			return !shouldPassCheck
+		default:
+			if cond() {
+				return shouldPassCheck
+			}
+			time.Sleep(interval)
+		}
+	}
```

Comment:
referencing a previous comment by @jimbishopp https://github.com/gravitational/teleport/pull/11845#discussion_r860327502

Error messages returned by `require.Eventually` and `require.Never` cannot be customized with variables that get updated during condition checks. So in order to keep using those functions we would have to lose some of the verbosity or alternatively come up with our own implementation.

Reply:
I think I would still prefer using `require.Eventually` rather than reimplementing it. Can always use `t.Log()` to log some extra context if the check fails.